MyNUnit/MyNUnit/MyNUnit.cs

```diff
@@ -0,0 +1,200 @@
+namespace MyNUnit;
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Reflection;
+using Attributes;
+using System;
+
+/// <summary>
+/// Class for running tests
+/// </summary>
+public class MyNUnit
+{
+    /// <summary>
+    /// Runs tests and outputs information
+    /// </summary>
+    public List<InformationAboutTest> Start(string path)
+    {
+        var result = new ConcurrentBag<InformationAboutTest>();
+        var dllFiles = Directory.GetFiles(path, "*.dll");
+        var instance = new object();
+        Parallel.ForEach(dllFiles, file =>
+        {
+            var types = Assembly.LoadFrom(file).GetTypes();
+            Parallel.ForEach(types, type =>
+            {
+                var listOfMethods = SplitMethodsIntoAttributes(type);
+                try
+                {
+                    RunAnyMethods(listOfMethods.BeforeClass, null);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"BeforeClass method returned exception ({e.Message}. Tests in class won't be run.");
+                    return;
+                }
+
+                Parallel.ForEach (listOfMethods.Test, test =>
+                {
+                    try
+                    {
+                        instance = Activator.CreateInstance(type);
+                    }
+                    catch(Exception e)
```

Comment:
```suggestion
                    catch (Exception e)
```
И тут стоило бы ловить более специфичное исключение. TargetInvocationException, скорее всего.